### PR TITLE
IFS-1887: Anpassung der Nutzungsvorgaben isy-security

### PR DESCRIPTION
**Pull request by FabianPerl**
Depends on #42 & #46  
Issue: ISY-254

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # 3.0.0
+- `IFS-1887`: [isyfact-standards-doc] Ergänzung der Nutzungsvorgaben für neuen Baustein Sicherheit
```
Comment (huebnercl) [inline CHANGELOG.md]:
```suggestion
- `IFS-1887`: [isyfact-standards-doc] Ergänzung der Nutzungsvorgaben für neuen Sicherheitsbaustein `isy-security`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +51,13 @@
 - `IFS-2153`: [isyfact-task] Umstellung isy-task auf Spring Boot:
   #### _BREAKING CHANGE: Einführung von Spring Boot_
   * Taskkonfiguration und TaskkonfigurationVerwalter entfernt, Übersetzungen hinzugefügt
+- `IFS-2416`: [isyfact-standards-bom] zentrale Versionsverwaltung von `isy-security`
+- `IFS-2416`: [isy-task] Umstellung von isy-sicherheit auf isy-security
+  #### _Breaking Change:_
+  * Die Konfiguration von `benutzer`, `passwort` und `bhknz` erfolgt über isy-security ClientRegistrations und für IsyTaskConfigurationProperties muss nur eine `registrationId` zur Authentifizierung konfiguriert werden.
+- `IFS-1853`: [isy-batchrahmen] Umstellung von `isy-sicherheit` und `isy-aufrufkontext` auf neuen Baustein `isy-security`
+  #### _Breaking Change:_
+  * Die Konfiguration von `benutzer`, `passwort` und `bhknz` erfolgt über isy-security ClientRegistrations. Batches müssen nur noch eine `registrationId` zur Authentifizierung bereitstellen. Die Methode `getAuthenticationCredentials` der `BatchAusfuehrungsBean` wurde entfernt.
```
Comment (huebnercl) [inline CHANGELOG.md]: Missing comments about ISY-147 changes with additional methods in AuthentifizierungsManager of isy-security for application-usage without 2AF by BHKNZ

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 3]
```diff
@@ -759,19 +749,15 @@ Folgende Informationen zum Benutzer sind anzugeben
 [cols="2,8",options="header"]
 |====
 |Property |Beschreibung
-|`Bhknz` |Das Behörden- oder Organisationskennzeichen, der Behörde oder Organisation, der der Benutzer angehört.
-|`Benutzer` |Die Kennung des technischen Benutzers (internes System)
-|`Passwort` |Das Kennwort des Benutzers
+|`registrationId` |Die ID der Client Registration aus `isy-security`, die zur Authentifizierung genutzt wird
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: missing comments about changes with ISY-147, especially here the renaming of the parameter `oaut2ClientRegistrationId`

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 4]
```diff
@@ -759,19 +749,15 @@ Folgende Informationen zum Benutzer sind anzugeben
...
 
 In der Regel wird der Benutzer aus der betrieblichen Konfiguration der Anwendung oder alternativ aus den Aufrufparametern beim Batchstart gelesen.
 
 In Ausnahmefällen ist es auch möglich, einen Batch zu implementieren, der ohne Benutzer laufen soll.
 Dies ist nur möglich, wenn bei Aufrufen des Anwendungskerns keine Autorisierungsprüfungen stattfinden und auch keine Nachbarsystemschnittstellen aufgerufen werden.
-Soll der Batch ohne Benutzer laufen, so ist in der Methode `null` zurückzugeben.
 
-Enthält eine Anwendung mehrere Batches, so enthält sie auch mehrere BatchAusführungsBeans und kann jeden Batch mit einem
-eigenen Benutzer ausstatten.
+Enthält eine Anwendung mehrere Batches, so enthält sie auch mehrere BatchKonfigurationen und kann jeden Batch mit einer eigenen `registrationId` ausstatten.
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: renaming of parameter `registrationId` based on ISY-147

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 5]
```diff
@@ -1064,11 +1050,10 @@ NOTE: Falls (in einem ungewöhnlichen Sonderfall) mit `ResultSet`-Iteratoren gea
...
 
 [[informationen-zum-batch-benutzer-bereitstellen]]
-=== Informationen zum Batch-Benutzer bereitstellen
+=== Informationen zur Authentifizierung bereitstellen
 
-Batches müssen beim Start einen Benutzer authentifizieren und autorisieren, bevor der fachliche Teil der Batchverarbeitung starten kann.
-Der Batchrahmen fordert von der Ausführungsbean die Bereitstellung der Informationen zum Benutzer, um diesen über die T-Komponente Sicherheit zu authentifizieren.
-Dazu ist für das Ausführungsbean die Methode `getAuthenticationCredentials()` umzusetzen.
+Batches müssen beim Start authentifiziert und autorisiert werden, bevor der fachliche Teil der Batchverarbeitung starten kann.
+Der Batchrahmen fordert von der BatchKonfiguration die Bereitstellung der `registrationId` zur ClientRegistration, um den Batch über den Baustein `isy-security` zu authentifizieren.
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: renaming of parameter `registrationId` based on ISY-147

File: isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/changelog.adoc [hunk 6]
```diff
@@ -5,11 +5,13 @@
...
-- `IFS-1852` Kapitel zur Authentifizierung eines Clients mit Client Credentials hinzugefügt
-- `IFS-2403` Abschnitt zum OpenId Connect Provider Mock aus isy-security-test hinzugefügt
+- `IFS-2302`: Rollen/Rechte-Konfiguration hinzugefügt
+- `IFS-1850`: Einleitung zu den Nutzungsvorgaben Security hinzugefügt
+- `IFS-1852`: Kapitel zur Authentifizierung eines Clients mit Client Credentials hinzugefügt
+- `IFS-2403`: Abschnitt zum OpenId Connect Provider Mock aus isy-security-test hinzugefügt
+- `IFS-1854`: Umsetzung des Berechtigungsmanagers
 - `IFS-1884` Umsetzung: Methodensicherheit
 - `IFS-1959` Abschnitt zum Auslesen des Behördenkennzeichens aus Security Context hinzugefügt
+- `IFS-1960` Kapitel zur Authentifizierung eines Clients mit Resource Owner Password Credentials hinzugefügt
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/changelog.adoc]: Missing comments about ISY-147 changes with additional methods in `AuthentifizierungsManager` of isy-security for application-usage without 2AF by BHKNZ

File: isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/inhalt.adoc [hunk 7]
```diff
@@ -2,19 +2,77 @@
...
+
+Die Software-Architektur des Security-Bausteins bildet das Fundament für die Sicherheit von Anwendungen in modernen Softwarelösungen.
+Die korrekte Gestaltung und Implementierung dieser Architektur ist entscheidend, um Schutzmechanismen gegen potenzielle Sicherheitsbedrohungen zu etablieren.
+In diesem Kapitel wird detailliert auf die verschiedenen Aspekte der Software-Architektur des Security-Bausteins eingegangen und bewährte Methoden sowie bewährte Praktiken vorgestellt, um eine robuste und zuverlässige Sicherheitsinfrastruktur zu schaffen.
+
+[[security-interface]]
+=== Security
+Das Interface `Security` stellt die zentrale Schnittstelle bereit, über die auf den Baustein zugegriffen wird.
+
+NOTE: Es empfiehlt sich, die Beans direkt zu autowiren und das Interface hauptsächlich für eine Migration von isy-sicherheit zu verwenden.
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/inhalt.adoc]:
```suggestion
NOTE: Es empfiehlt sich, die Beans direkt zu 'autowiren' und das Interface hauptsächlich für eine Migration vom alten Sicherheitsbaustein `isy-sicherheit` auf den neuen Baustein `isy-security` zu verwenden.
```

File: isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/inhalt.adoc [hunk 8]
```diff
@@ -2,19 +2,77 @@
...
+
+Die folgenden Methoden bieten Zugriff auf eine Liste von allen im System hinterlegten Rollen, den Berechtigungsmanager und den optionalen Authentifizierungsmanager:
+
+`Set<String> getAlleRollen()`:: Liefert die Menge aller im System konfigurierten Rollen.
+`Berechtigungsmanager getBerechtigungsManager()`:: Stellt eine Referenz auf das `Berechtigungsmanager` Bean bereit.
+`Optional<Authentifizierungsmanager> getAuthentifizierungsManager()`:: Liefert ein `Optional`, das eine Referenz auf den `Authentifizierungsmanager` enthält, oder `null` falls die Anwendung als Resource-Server agiert und kein `Authentifizierungsmanager` benötigt wird.
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/inhalt.adoc:24]: The Type `Optional` was removed with ISY-147, right?

**Review by huebnercl — CHANGES_REQUESTED** [review]
Please update and merge this branch with master-branch including the Ticket `ISY-147` with the new and approved isy-security component and adapt the documentation before further review actions

File: isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/inhalt.adoc (hunk 8 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/nutzungsvorgaben/inhalt.adoc:24]: As far as I can see, [the return value is still Optional.](https://github.com/IsyFact/isyfact-standards/blob/master/isy-security/src/main/java/de/bund/bva/isyfact/security/core/Security.java#L26)

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc (hunk 5 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: Variable was adjusted to `oauth2ClientRegistrationId` after rebase to master (including ticket ISY-147)

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc (hunk 4 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: Variable was adjusted to `oauth2ClientRegistrationId` after rebase to master (including Ticket ISY-147)

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 9]
```diff
@@ -738,19 +738,9 @@ Datei `JRE_HOME/lib/management/jmxremote.access` readwrite-Zugriff erlaubt ist.
 Ein Batch wird innerhalb der Plattform gestartet.
 Dazu verwendet der Batch die Benutzerkennung eines technischen Benutzers „internes System“, die in seiner Startkonfiguration (als Konfigurationsdatei) hinterlegt und im Benutzerverzeichnis vorhanden ist.
 
-Der Batchrahmen nutzt die T-Komponente Sicherheit (siehe xref:isy-sicherheit:nutzungsvorgaben/master.adoc#einleitung[Nutzungsvorgaben Sicherheit]), um den anfragenden Benutzer über den IAM-Service der Plattform zu authentifizieren.
-Das IAM-Service authentifiziert den Benutzer anhand der angegebenen Benutzerkennung und des Passwortes, registriert alle Informationen zum Benutzer in der T-Komponente AufrufKontextVerwalter und schließt die Session des Benutzers im IAM-System.
+Der Batchrahmen verwendet den Baustein `isy-security`, um den anfragenden Benutzer über einen der Authentifizierungswege Client Credential oder Ressource Owner Password (deprecated) via `oauth2ClientRegistrationId` zu authentifizieren.
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]:
```suggestion
Der Batchrahmen verwendet den Baustein `isy-security`, um den in der Batch-Konfiguration via `oauth2ClientRegistrationId` definierten Benutzer über einen der Authentifizierungswege `Client Credential` oder `Ressource Owner Password Credential` (deprecated) zu authentifizieren.
```

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 10]
```diff
@@ -759,19 +749,15 @@ Folgende Informationen zum Benutzer sind anzugeben
 [cols="2,8",options="header"]
 |====
 |Property |Beschreibung
-|`Bhknz` |Das Behörden- oder Organisationskennzeichen, der Behörde oder Organisation, der der Benutzer angehört.
-|`Benutzer` |Die Kennung des technischen Benutzers (internes System)
-|`Passwort` |Das Kennwort des Benutzers
+|`oauth2ClientRegistrationId` |Die ID der OAuth 2.0 Client Registration aus `isy-security`, die zur Authentifizierung genutzt wird
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]:
```suggestion
|`oauth2ClientRegistrationId` |Die ID der OAuth 2.0 Client Registration aus `isy-security`, die zur Authentifizierung genutzt wird. Die notwendigen Properties die für eine gültige `ClientRegistration` definiert werden müssen, sind in den Nutzungsvorgaben der Komponente `isy-security` beschrieben.
```

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 11]
```diff
@@ -1064,11 +1050,10 @@ NOTE: Falls (in einem ungewöhnlichen Sonderfall) mit `ResultSet`-Iteratoren gea
...
 
 [[informationen-zum-batch-benutzer-bereitstellen]]
-=== Informationen zum Batch-Benutzer bereitstellen
+=== Informationen zur Authentifizierung bereitstellen
 
-Batches müssen beim Start einen Benutzer authentifizieren und autorisieren, bevor der fachliche Teil der Batchverarbeitung starten kann.
-Der Batchrahmen fordert von der Ausführungsbean die Bereitstellung der Informationen zum Benutzer, um diesen über die T-Komponente Sicherheit zu authentifizieren.
-Dazu ist für das Ausführungsbean die Methode `getAuthenticationCredentials()` umzusetzen.
+Batches müssen beim Start authentifiziert und autorisiert werden, bevor der fachliche Teil der Batchverarbeitung starten kann.
+Der Batchrahmen fordert von der BatchKonfiguration die Bereitstellung der `oauth2ClientRegistrationId` zur ClientRegistration, um den Batch über den Baustein `isy-security` zu authentifizieren.
```
Comment (huebnercl) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]:
```suggestion
Der Batchrahmen fordert, dass in der BatchKonfiguration zur Batch spezifischen `Batch-ID` das neue Property `oauth2ClientRegistrationId` hinzugefügt wird, dessen Wert als Indirektion auf eine `ClientRegistration` führt, um den Batch über den Baustein `isy-security` zu authentifizieren.
```

**Review by huebnercl — CHANGES_REQUESTED** [review]
the solution is quite near, simply some small reformulations...

**Comment by FabianPerl** [comment]
@huebnercl 

I rebased the branch and commited the requested changes. Hope it fits!

**Review by huebnercl — APPROVED** [review]

**Pull request merged**